### ISSUE TITLE
what this code do

### ISSUE THREAD

**Issue by Luckygyana** [HobbitLong/SupContrast#11]
for idx, (images, labels) in enumerate(train_loader):
        data_time.update(time.time() - end)

        images = torch.cat([images[0], images[1]], dim=0)

**Comment by HobbitLong** [comment]
`image[0]` is the first set of crops for each batch: `[bsz, 3, h, w]`;
`image[1]` is the second set of crops for each batch: `[bsz, 3, h, w]`.
You need to concatenate them in the batch dimension to extract features in a single forward pass.

**Comment by StackChan** [comment]
> `image[0]` is the first set of crops for each batch: `[bsz, 3, h, w]`; `image[1]` is the second set of crops for each batch: `[bsz, 3, h, w]`. You need to concatenate them in the batch dimension to extract features in a single forward pass.

but why images is [2,bsz,3,h,w] instead of [bsz,2,3,h,w]? .I hope you know what I mean, since 'bsz = labels.shape[0]'.
let's go back to the code,
`    for idx, (images, labels) in enumerate(train_loader):
        data_time.update(time.time() - end)

        images = torch.cat([images[0], images[1]], dim=0)
        if torch.cuda.is_available():
            images = images.cuda(non_blocking=True)
            labels = labels.cuda(non_blocking=True)
        bsz = labels.shape[0]

        # warm-up learning rate
        warmup_learning_rate(opt, epoch, idx, len(train_loader), optimizer)
`
